src/qt/receivecoinsdialog.cpp

```diff
@@ -100,11 +100,19 @@ void ReceiveCoinsDialog::setModel(WalletModel *_model)
             ui->useBech32->setCheckState(Qt::Unchecked);
         }
 
-        // eventually disable the main receive button if private key operations are disabled
-        ui->receiveButton->setEnabled(!model->privateKeysDisabled());
+        // Set the button to be enabled or disabled based on whether the wallet can give out new addresses.
+        ui->receiveButton->setEnabled(model->canGetAddresses());
+
+        // Enable/disable the receive button if the wallet is now able/unable to give out new addresses.
+        connect(_model, &WalletModel::notifyCanGetAddressesChanged, this, &ReceiveCoinsDialog::updateReceiveButton);
```

Comment:
I think you could avoid the slot, as it is only used once:
```cpp
connect(model, &WalletModel::canGetAddressesChanged, [this] {
    ui->receiveButton->setEnabled(model->canGetAddresses());
});
```

Reply:
Changed.